File: smart-dialog/src/main/java/com/coder/zzq/smartshow/dialog/ChooseListAdapter.java
package com.coder.zzq.smartshow.dialog;

import android.graphics.drawable.Drawable;
import android.graphics.drawable.StateListDrawable;
import android.support.annotation.ColorInt;
import android.support.v4.graphics.drawable.DrawableCompat;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.CheckedTextView;

import com.coder.zzq.smartshow.core.Utils;

import java.util.ArrayList;
import java.util.List;

public class ChooseListAdapter extends BaseAdapter {
    private List mItems = new ArrayList();
    private int mCheckMarkPos = Gravity.LEFT;
    @ColorInt
    private int mCheckMarkColor = Utils.getColorFromRes(R.color.colorPrimary);

    private boolean mUseCubeMark;

    public void setItems(List items) {
        mItems.clear();
        mItems.addAll(items);
        notifyDataSetChanged();
    }

    public void setCheckMarkPos(int checkMarkPos) {
        if (mCheckMarkPos != checkMarkPos) {
            mCheckMarkPos = checkMarkPos;
            notifyDataSetChanged();
        }
    }

    public void setCheckMarkColor(@ColorInt int checkMarkColor) {
        if (mCheckMarkColor != checkMarkColor) {
            mCheckMarkColor = checkMarkColor;
            notifyDataSetChanged();
        }
    }

    public void setUseCubeMark(boolean useCubeMark) {
        if (mUseCubeMark != useCubeMark) {
            mUseCubeMark = useCubeMark;
            notifyDataSetChanged();
        }
    }

    @Override
    public int getCount() {
        return mItems.size();
    }

    @Override
    public Object getItem(int position) {
        return mItems.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        CheckedTextView itemView = (CheckedTextView) convertView;
        if (itemView == null) {
            itemView = (CheckedTextView) LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_single_choice, parent, false);
        }
        StateListDrawable stateListDrawable = new StateListDrawable();
        Drawable checkedDrawable = Utils.getDrawableFromRes(mUseCubeMark ? R.drawable.smart_show_multiple_choose : R.drawable.smart_show_single_choose);
        DrawableCompat.setTint(checkedDrawable, mCheckMarkColor);
        stateListDrawable.addState(
                new int[]{
                        android.R.attr.state_checked
                },
                checkedDrawable
        );
        Drawable uncheckedDrawable = Utils.getDrawableFromRes(mUseCubeMark ? R.drawable.smart_show_multiple_unchoose : R.drawable.smart_show_single_unchoose);
        uncheckedDrawable.setBounds(0, 0, Utils.dpToPx(5), Utils.dpToPx(5));
        stateListDrawable.addState(
                new int[]{
                        -android.R.attr.state_checked
                },
                uncheckedDrawable
        );
        stateListDrawable.setBounds(0, 0, Utils.dpToPx(17), Utils.dpToPx(17));
        Drawable leftDrawable = null;
        Drawable rightDrawable = null;
        if (mCheckMarkPos == Gravity.LEFT) {
            leftDrawable = stateListDrawable;
        } else {
            rightDrawable = stateListDrawable;
        }
        itemView.setCompoundDrawables(leftDrawable, null, rightDrawable, null);
        Object item = mItems.get(position);
        itemView.setText(item instanceof StringItem ? ((StringItem) item).getString() : item.toString());
        return itemView;
    }
}

File: smart-dialog/src/main/java/com/coder/vincent/smart_dialog/choose_list/ChooseListAdapter.kt
package com.coder.vincent.smart_dialog.choose_list

import android.graphics.drawable.Drawable
import android.graphics.drawable.StateListDrawable
import android.view.Gravity
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.CheckedTextView
import androidx.core.graphics.drawable.DrawableCompat
import com.coder.vincent.series.common_lib.bean.TextStyle
import com.coder.vincent.series.common_lib.dpToPx
import com.coder.vincent.series.common_lib.layoutInflater
import com.coder.vincent.series.common_lib.resourceToColor
import com.coder.vincent.series.common_lib.resourceToDrawable
import com.coder.zzq.smartshow.dialog.R
import com.coder.zzq.smartshow.dialog.databinding.ListItemChoiceBinding

class ChooseListAdapter : BaseAdapter() {
    private var items: List<String> = listOf()
    private var iconStyle: Int = ICON_STYLE_CIRCLE
    private var itemCenter: Boolean = true
    private var itemLabelStyle: TextStyle? = null
    private var iconColor: Int = R.color.colorPrimary.resourceToColor()
    private var iconPosition: Int = ICON_POSITION_LEFT

    fun setItems(items: List<String>, notify: Boolean = true) {
        this.items = items
        if (notify) {
            notifyDataSetChanged()
        }
    }

    fun setIconStyle(style: Int, notify: Boolean = true) {
        this.iconStyle = style
        if (notify) {
            notifyDataSetChanged()
        }
    }

    fun setItemCenter(itemCenter: Boolean, notify: Boolean = true) {
        this.itemCenter = itemCenter
        if (notify) {
            notifyDataSetChanged()
        }
    }

    fun setItemLabelStyle(textStyle: TextStyle, notify: Boolean = true) {
        this.itemLabelStyle = textStyle
        if (notify) {
            notifyDataSetChanged()
        }
    }

    fun setIconColor(iconColor: Int, notify: Boolean = true) {
        this.iconColor = iconColor
        if (notify) {
            notifyDataSetChanged()
        }
    }

    fun setIconPosition(position: Int, notify: Boolean = true) {
        this.iconPosition = position
        if (notify) {
            notifyDataSetChanged()
        }
    }

    override fun getCount(): Int {
        return items.size
    }

    override fun getItem(position: Int): String {
        return items[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val itemView = convertView ?: ListItemChoiceBinding.inflate(
            layoutInflater,
            parent,
            false
        ).root
        return (itemView as CheckedTextView).apply {
            gravity = if (itemCenter) Gravity.CENTER else Gravity.LEFT or Gravity.CENTER_VERTICAL
            text = items[position]
            itemLabelStyle?.let {
                setTextColor(it.color)
                paint.isFakeBoldText = it.bold
                textSize = it.size
            }
            val stateListDrawable = StateListDrawable()
            val checkedDrawable =
                if (iconStyle == ICON_STYLE_CIRCLE)
                    R.drawable.smart_show_circle_choose.resourceToDrawable()
                else
                    R.drawable.smart_show_cube_choose.resourceToDrawable()
            DrawableCompat.setTint(checkedDrawable!!, iconColor)
            stateListDrawable.addState(intArrayOf(android.R.attr.state_checked), checkedDrawable)
            val uncheckedDrawable =
                if (iconStyle == ICON_STYLE_CIRCLE)
                    R.drawable.smart_show_circle_unchoose.resourceToDrawable()
                else
                    R.drawable.smart_show_cube_unchoose.resourceToDrawable()
            stateListDrawable.addState(intArrayOf(0), uncheckedDrawable)
            stateListDrawable.setBounds(0, 0, 17f.dpToPx(), 17f.dpToPx())
            var leftDrawable: Drawable? = null
            var rightDrawable: Drawable? = null
            if (iconPosition == ICON_POSITION_LEFT) {
                leftDrawable = stateListDrawable
            } else {
                rightDrawable = stateListDrawable
            }
            setCompoundDrawables(leftDrawable, null, rightDrawable, null)
        }
    }
}